### ISSUE TITLE
Factored methods do not work in CoreNLP 3.6.0

### ISSUE THREAD

**Issue by reckart** [stanfordnlp/CoreNLP#195]
Factored methods do not work in CoreNLP 3.6.0. 

The difference appears to come from the way that 3.6.0 and 3.5.2 invoke the parser inside the ParserAnnotator.

3.5.2: 

```
  tree = pq.getBestParse(); 
```

3.6.0:

```
  // todo [cdm 2015]: This should just use bestParse method if only getting 1 best parse.
  List<ScoredObject<Tree>> scoredObjects = pq.getKBestPCFGParses(this.kBest);
```

I checked and pq.getKBestPCFGParses() actually returns the same results for 3.6.0 and 3.5.2.

But apparently getBestParse() does not produce the same result as getting the getKBestPCFGParse() with the highest score.

Full details can be found in this thread on the mailing list: 

https://mailman.stanford.edu/pipermail/java-nlp-user/2016-January/007451.html

**Comment by reckart** [comment]
Seems to have been fixed in https://github.com/stanfordnlp/CoreNLP/commit/4910f838ff7b8ee7036427afb5d0381158e23cf0

Unfortunately, no release yet with this fixed.

**Comment by reckart** [comment]
Appears to be fixed in CoreNLP 3.7.0.
